src/test/test_snap_mapper.cc

```diff
@@ -535,27 +554,22 @@ class MapperVerifier {
     return mapper->make_purged_snap_key(std::forward<Args>(args)...);
   }
 
-  void trim_snap() {
-    std::lock_guard l{lock};
-    if (snap_to_hobject.empty())
-      return;
-    map<snapid_t, set<hobject_t> >::iterator snap =
-      rand_choose(snap_to_hobject);
-    set<hobject_t> hobjects = snap->second;
-
+  // must be called with lock held
```

Comment:
Can a lock_guard be used from within the function instead of the comment?
Same in trim_snap_force() and __add_object()

Reply:
not sure what you mean

Reply:
Instead of adding the comment can we simply call the lock from within the function to enforce that the lock is indeed held?

Reply:
Do you suggest mechanism which can verify that a lock is held?

Reply:
> Do you suggest mechanism which can verify that a lock is held?

No, I meant simply taking the lock inside the function's scope.

Reply:
the lock protects a bigger scope than this function
It needs to protect any access to data-members (done from multiple locations inside and outside the function)